File: app/src/main/java/me/tysheng/xishi/base/BaseActivity.java
package me.tysheng.xishi.base;

import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.view.Window;

import com.baidu.mobstat.StatService;
import com.trello.rxlifecycle.components.support.RxAppCompatActivity;

import me.tysheng.xishi.App;
import me.tysheng.xishi.dagger.component.ActivityComponent;
import me.tysheng.xishi.dagger.component.ApplicationComponent;
import me.tysheng.xishi.dagger.component.DaggerActivityComponent;
import me.tysheng.xishi.dagger.module.ActivityModule;

/**
 * Created by shengtianyang on 16/7/11.
 */
public abstract class BaseActivity extends RxAppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            getWindow().requestFeature(Window.FEATURE_CONTENT_TRANSITIONS);
        }
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onPause() {
        super.onPause();
        StatService.onPause(this);
    }

    @Override
    protected void onResume() {
        super.onResume();
        StatService.onResume(this);
    }

    protected ApplicationComponent getApplicationComponent() {
        return ((App) getApplication()).getApplicationComponent();
    }

    protected ActivityComponent injectAppComponent() {
        return DaggerActivityComponent.builder()
                .applicationComponent(getApplicationComponent())
                .activityModule(getActivityModule())
                .build();
    }

    protected ActivityModule getActivityModule() {
        return new ActivityModule(this);
    }

    /**
     * Fragment之间的切换
     *
     * @param from 当前
     * @param to   目标
     * @param id
     * @param tag
     */
    protected void jumpFragment(Fragment from, Fragment to, int id, String tag) {
        FragmentManager manager = getSupportFragmentManager();
        if (to == null) {
            return;
        }
        FragmentTransaction transaction = manager.beginTransaction();
        if (from == null) {
            transaction.add(id, to, tag);
        } else {
            transaction.hide(from);
            if (to.isAdded()) {
                transaction.show(to);
            } else {
                transaction.add(id, to, tag);
            }
        }
        transaction
                .setCustomAnimations(0, 0, android.R.anim.fade_in, android.R.anim.fade_out)
                .commitAllowingStateLoss();

    }

}

File: app/src/main/java/me/tysheng/xishi/base/BaseActivity.kt
package me.tysheng.xishi.base

import com.trello.rxlifecycle2.components.support.RxAppCompatActivity
import me.tysheng.xishi.App
import me.tysheng.xishi.dagger.component.ActivityComponent
import me.tysheng.xishi.dagger.component.ApplicationComponent
import me.tysheng.xishi.dagger.component.DaggerActivityComponent
import me.tysheng.xishi.dagger.module.ActivityModule

/**
 * Created by shengtianyang on 16/7/11.
 */
abstract class BaseActivity : RxAppCompatActivity() {

    private val applicationComponent: ApplicationComponent?
        get() = (application as App).applicationComponent

    private val activityModule: ActivityModule
        get() = ActivityModule(this)

    protected fun injectAppComponent(): ActivityComponent {
        return DaggerActivityComponent.builder()
                .applicationComponent(applicationComponent)
                .activityModule(activityModule)
                .build()
    }

}
